crates/ruff_python_ast/src/identifier.rs

```diff
@@ -0,0 +1,585 @@
+//! Extract [`TextRange`] information from AST nodes.
+//!
+//! In the `RustPython` AST, each node has a `range` field that contains the
+//! start and end byte offsets of the node. However, attributes on those
+//! nodes may not have their own ranges. In particular, identifiers are
+//! not given their own ranges, unless they're part of a name expression.
+//!
+//! For example, given:
+//! ```python
+//! def f():
+//!     ...
+//! ```
+//!
+//! The statement defining `f` has a range, but the identifier `f` does not.
+//!
+//! This module assists with extracting [`TextRange`] ranges from AST nodes
+//! via manual lexical analysis.
```

Comment:
This file got collapsed, but it's really the one that needs to be reviewed. Most of the rest is just API migration.